lib/iris/fileformats/netcdf.py

```diff
@@ -455,85 +429,78 @@ def __setstate__(self, state):
 
 
 def _assert_case_specific_facts(engine, cf, cf_group):
-    # Initialise pyke engine "provides" hooks.
-    # These are used to patch non-processed element attributes after rules activation.
+    # Initialise a data store for built cube elements.
+    # This is used to patch element attributes *not* setup by the actions
+    # process, after the actions code has run.
     engine.cube_parts["coordinates"] = []
     engine.cube_parts["cell_measures"] = []
     engine.cube_parts["ancillary_variables"] = []
 
     # Assert facts for CF coordinates.
     for cf_name in cf_group.coordinates.keys():
-        engine.add_case_specific_fact(
-            _PYKE_FACT_BASE, "coordinate", (cf_name,)
-        )
+        engine.add_case_specific_fact("coordinate", (cf_name,))
 
     # Assert facts for CF auxiliary coordinates.
     for cf_name in cf_group.auxiliary_coordinates.keys():
-        engine.add_case_specific_fact(
-            _PYKE_FACT_BASE, "auxiliary_coordinate", (cf_name,)
-        )
+        engine.add_case_specific_fact("auxiliary_coordinate", (cf_name,))
 
     # Assert facts for CF cell measures.
     for cf_name in cf_group.cell_measures.keys():
-        engine.add_case_specific_fact(
-            _PYKE_FACT_BASE, "cell_measure", (cf_name,)
-        )
+        engine.add_case_specific_fact("cell_measure", (cf_name,))
 
     # Assert facts for CF ancillary variables.
     for cf_name in cf_group.ancillary_variables.keys():
-        engine.add_case_specific_fact(
-            _PYKE_FACT_BASE, "ancillary_variable", (cf_name,)
-        )
+        engine.add_case_specific_fact("ancillary_variable", (cf_name,))
 
     # Assert facts for CF grid_mappings.
     for cf_name in cf_group.grid_mappings.keys():
-        engine.add_case_specific_fact(
-            _PYKE_FACT_BASE, "grid_mapping", (cf_name,)
-        )
+        engine.add_case_specific_fact("grid_mapping", (cf_name,))
 
     # Assert facts for CF labels.
     for cf_name in cf_group.labels.keys():
-        engine.add_case_specific_fact(_PYKE_FACT_BASE, "label", (cf_name,))
+        engine.add_case_specific_fact("label", (cf_name,))
 
     # Assert facts for CF formula terms associated with the cf_group
     # of the CF data variable.
-    formula_root = set()
+
+    # Collect varnames of formula-root variables as we go.
+    # NOTE: use dictionary keys as an 'OrderedDict'
```

Comment:
```suggestion
    # NOTE: use dictionary keys as an 'OrderedSet'
```

The 'OrderedDict' is a real thing.
I **_meant_** to say 'OrderedSet' here, which is a missing concept.